core/services/relay/evm/types/types.go

```diff
@@ -25,6 +26,7 @@ import (
 type ChainWriterConfig struct {
 	Contracts    map[string]*ContractConfig
 	SendStrategy txmgrtypes.TxStrategy
+	MaxGasPrice  *mathBig.Int
```

Comment:
Why do we need a new field for this, given that we already have the ability to override per chain and per account?

Reply:
The existing config API consists of a func for looking up by account dynamically. Doesn't that need to be supported? https://github.com/smartcontractkit/chainlink/blob/develop/core/chains/evm/config/chain_scoped_gas_estimator.go#L17

Reply:
Again, this functionality mimics the [CCIP gas esimtation](https://github.com/smartcontractkit/ccip/blob/4a6f68e052f1e55596b1f8238f23806a7128c541/core/services/ocr2/plugins/ccip/prices/exec_price_estimator.go#L23). I suppose we could remove it entirely and rely on the gas estimator's config instead, is that what you are suggesting?

Reply:
I don't understand well enough what you are trying to achieve to say what should be happening. Do you intend to use this variable anywhere? I do not see it being set anywhere in this PR.

Reply:
@jmank88 Sorry - for some reason I didn't see the latter two comments when I initially responded. I'm not sure I understand your question. When a product that is using the ChainWriter needs the ChainWriter to be capable of gas estimation, it will include a `MaxGasPrice` in the config that is passed into `NewChainWriterService`. I'm not familiar with the `PriceMaxKey` function you shared but I supposed `MaxGasPrice` in the ChainWriter would be used to override that value in the gas estimator's `GetFee` function. For example the EVM gas estimators take the [min between the user defined max and the one specified in the gas estimators config. ](https://github.com/smartcontractkit/chainlink/blob/4fdfffdb4c4a6f16fcdb388fe38642398ce130da/core/chains/evm/gas/models.go#L510) So here, we are just enabling that override functionality. Does that make sense or am I failing to grasp something here?

Reply:
The reason it isn't being set anywhere in this PR is because the [Keystone Write Target](https://github.com/smartcontractkit/chainlink/blob/f0d2ca52ead9ba3136bafc8e2c5cf7915f858ee3/core/services/relay/evm/write_target.go#L57-L73) doesn't need gas estimation. When CCIP implements the ChainWriter they will have to configure `MaxGasPrice` in the `ChainWriterConfig` to enable gas esitmation.

Reply:
The max gas price is currently something that is configured per account, and thus completely out of the control of the sender of a transaction. When you send through a particular account, you are implicitly opting in to that account's maximum gas price. I don't understand how what we are doing here will connect to the existing system, and if we don't actually have a use for this code right now, why are we building it?

Reply:
> The max gas price is currently something that is configured per account, and thus completely out of the control of the sender of a transaction. 
I don't understand this statement, since [GetFee](https://github.com/smartcontractkit/chainlink/blob/4fdfffdb4c4a6f16fcdb388fe38642398ce130da/core/chains/evm/gas/models.go#L38) accepts a `maxFeePrice`. This PR isn't changing that logic in any way.

> and if we don't actually have a use for this code right now, why are we building it
Because this functionality is required for CCIPv2 to implement the `ChainWriter`

Reply:
> I don't understand this statement, since [GetFee](https://github.com/smartcontractkit/chainlink/blob/4fdfffdb4c4a6f16fcdb388fe38642398ce130da/core/chains/evm/gas/models.go#L38) accepts a maxFeePrice. This PR isn't changing that logic in any way.

Here `GetFee` is called with a `keySpecificGasPrice` from tx attempt builder:
https://github.com/smartcontractkit/chainlink/blob/develop/core/chains/evm/txmgr/attempts.go#L60-L61

Reply:
> Because this functionality is required for CCIPv2 to use the ChainWriter

Do you mean that there will be a usage in the CCIP repo? please link any corresponding PRs

Reply:
> Here GetFee is called with a keySpecificGasPrice from tx attempt builder:
https://github.com/smartcontractkit/chainlink/blob/develop/core/chains/evm/txmgr/attempts.go#L60-L61

Ahh I see, thanks for clarifying. I wonder what address we could use here since `GetFeeComponents` is used outside of the context of a specific Tx / address. Maybe we need to require the caller to pass an address? 

> Do you mean that there will be a usage in the CCIP repo? please link any corresponding PRs

There are no PRs yet in the CCIP repo to my knowledge, I believe they are waiting for us to complete the EVM implementation of the ChainWriter.